src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -1608,15 +1610,29 @@ public IEnumerable<SourceLocation> FindBreakpointLocations(BreakpointRequest req
 
             foreach (MethodInfo method in sourceFile.Methods)
             {
-                if (!method.DebugInformation.SequencePointsBlob.IsNil)
+                if (method.DebugInformation.SequencePointsBlob.IsNil)
+                    continue;
+                if (!(method.StartLocation.Line <= request.Line && request.Line <= method.EndLocation.Line))
+                    continue;
+                foreach (SequencePoint sequencePoint in method.DebugInformation.GetSequencePoints())
                 {
-                    foreach (SequencePoint sequencePoint in method.DebugInformation.GetSequencePoints())
-                    {
-                        if (!sequencePoint.IsHidden && Match(sequencePoint, request.Line, request.Column))
-                            yield return new SourceLocation(method, sequencePoint);
-                    }
+                    if (sequencePoint.IsHidden)
+                        continue;
+                    if (Match(sequencePoint, request.Line, request.Column))
+                        locations.Add(new SourceLocation(method, sequencePoint));
+                    else if (ifNoneFoundThenFindNext && nextSeqPoint == null && sequencePoint.StartLine > request.Line)
+                        nextSeqPoint = new SourceLocation(method, sequencePoint);
```

Comment:
If either of these is true, then we can break out of the outer loop (for methods).